jme3-examples/src/main/java/jme3test/games/RollingTheMonkey.java

```diff
@@ -0,0 +1,417 @@
+        }
+        
+        cam.lookAt(player.getPhysicsLocation(), Vector3f.UNIT_Y);
+    }
+
+    @Override
+    public void onAction(String name, boolean isPressed, float tpf) {
+        switch(name) {
+            case INPUT_MAPPING_FORWARD:
+                keyForward = isPressed;
+                break;
+            case INPUT_MAPPING_BACKWARD:
+                keyBackward = isPressed;
+                break;
+            case INPUT_MAPPING_LEFT:
+                keyLeft = isPressed;
+                break;
+            case INPUT_MAPPING_RIGHT:
+                keyRight = isPressed;
+                break;
+            case INPUT_MAPPING_RESET:
+                enqueue(new Callable<Void>() {
+                    @Override
+                    public Void call() {
+                        reset();
+                        return null;
+                    }
+                });
+                break;
+        }
+    }
+    @Override
+    public void collision(PhysicsCollisionEvent event) {
+        Spatial nodeA = event.getNodeA();
+        Spatial nodeB = event.getNodeB();
+        
+        String nameA = nodeA == null ? "" : nodeA.getName();
+        String nameB = nodeB == null ? "" : nodeB.getName();
+        
+        if(nameA.equals("player") && nameB.startsWith("pickUp")) {
+            GhostControl pickUpControl = nodeB.getControl(GhostControl.class);
+            if(pickUpControl != null && pickUpControl.isEnabled()) {
+                pickUpControl.setEnabled(false);
+                nodeB.removeFromParent();
+                nodeB.setLocalScale(0.0f);
+                score += 1;
+                if(score >= PICKUP_COUNT) {
+                    messageText.setLocalScale(1.0f);
+                }
+            }
+        } else if(nameA.startsWith("pickUp") && nameB.equals("player")) {
+            GhostControl pickUpControl = nodeA.getControl(GhostControl.class);
+            if(pickUpControl != null && pickUpControl.isEnabled()) {
+                pickUpControl.setEnabled(false);
+                nodeA.setLocalScale(0.0f);
+                score += 1;
+                if(score >= PICKUP_COUNT) {
+                    messageText.setLocalScale(1.0f);
+                }
+            }
+        }
+    }
+    
+    private void reset() {
+        // Reset the pickups
+        for(Spatial pickUp : pickUps.getChildren()) {
+            GhostControl pickUpControl = pickUp.getControl(GhostControl.class);
+            if(pickUpControl != null) {
+                pickUpControl.setEnabled(true);
+            }
+            pickUp.setLocalScale(1.0f);
+        }
+        // Reset the player
+        player.setPhysicsLocation(PLAYER_START.clone());
+        player.setAngularVelocity(Vector3f.ZERO.clone());
+        player.setLinearVelocity(Vector3f.ZERO.clone());
+        // Reset the score
+        score = 0;
+        // Reset the message
+        messageText.setLocalScale(0.0f);
+    }
+    
+}

```

Comment:
to address concerns of mvvm style coding I could create a protected class here
```
class PlayerBallControl extends RigidBodyControl implements PhysicsCollisionListener {
	public PlayerBallControl(float radius, float mass) {
		super(new SphereCollisionShape(radius), mass);
	}
	@Override
	public void collision(PhysicsCollisionEvent event) {
		if(!this.equals(event.getSource())) {
			return;
		}
		// TODO: handle collision event
	}
}
```
Of course also creating separate classes for player input and pickups.